packages/server/routes/api/events.js

```diff
@@ -280,4 +281,58 @@ router.get(
   }
 );
 
+router.post(
+  "/:eventId/check-in",
+  AuthController.authenticate,
+  eventIdValidator,
+  userIdBodyValidator,
+  async (request, response) => {
+    const errors = validationResult(request);
+
+    if (!errors.isEmpty()) {
+      response.status(400).json({
+        status: "fail",
+        data: errors.array(),
+      });
+      return;
+    }
+
+    try {
+      const { status, userId } = request.body;
+      const eventId = request.params.eventId;
+      const event = await EventController.getEvent(eventId);
+
+      const currentTime = new Date();
+      const checkInStartTime = new Date(event.startDate.getTime() - 5 * 60000);
+      const checkInEndTime = new Date(event.endDate.getTime() + 15 * 60000);
```

Comment:
You should stray away from using "magic numbers", numbers that are directly in code. It makes the code less readable because if someone else reads this or if you come back to it again in the future, it's not immediately clear what these numbers mean.

To fix this, assign the numbers to variables with clear, meaningful names instead and then use those here.

You can read more about magic numbers [here](https://stackoverflow.com/questions/47882/what-are-magic-numbers-and-why-do-some-consider-them-bad).